base/MAPL_NewArthParser.F90

```diff
@@ -742,6 +744,136 @@ SUBROUTINE CopyScalarToField(ptrs,rn,rc)
+    CALL Replace ('**','^ ',Func)                            ! Exponent into 1-Char. format
+    CALL RemoveSpaces (Func,ipos)
+    j = 1
+    ParCnt = 0
+    lFunc = LEN_TRIM(Func)
+    step: DO
+       IF (j > lFunc) CALL ParseErrMsg (j, FuncStr, ipos)
+       c = Func(j:j)
+       !-- -------- --------- --------- --------- --------- --------- --------- -------
+       ! Check for valid operand (must appear)
+       !-- -------- --------- --------- --------- --------- --------- --------- -------
+       IF (c == '-' .OR. c == '+') THEN                      ! Check for leading - or +
+          j = j+1
+          IF (j > lFunc) THEN 
+             CALL ParseErrMsg (j, FuncStr, ipos, 'Missing operand')
+             _ASSERT(.FALSE.,'needs informative message')
+          END IF
+          c = Func(j:j)
+          IF (ANY(c == Ops)) THEN
+             CALL ParseErrMsg (j, FuncStr, ipos, 'Multiple operators')
+             _ASSERT(.FALSE.,'needs informative message')
+          END IF
+       END IF
+       n = MathFunctionIndex (Func(j:))
+       IF (n > 0) THEN                                       ! Check for math function
+          j = j+LEN_TRIM(Funcs(n))
+          IF (j > lFunc) THEN 
+             CALL ParseErrMsg (j, FuncStr, ipos, 'Missing function argument')
+             _ASSERT(.FALSE.,'needs informative message')
+          END IF
+          c = Func(j:j)
+          IF (c /= '(') THEN 
+             CALL ParseErrMsg (j, FuncStr, ipos, 'Missing opening parenthesis')
+             _ASSERT(.FALSE.,'needs informative message')
+          END IF
+       END IF
+       IF (c == '(') THEN                                    ! Check for opening parenthesis
+          ParCnt = ParCnt+1
+          j = j+1
+          CYCLE step
+       END IF
+       IF (SCAN(c,'0123456789.') > 0) THEN                   ! Check for number
+          r = RealNum (Func(j:),ib,in,err)
+          IF (err) THEN
+             CALL ParseErrMsg (j, FuncStr, ipos, 'Invalid number format:  '//Func(j+ib-1:j+in-2))
+             _ASSERT(.FALSE.,'needs informative message')
+          END IF
+          j = j+in-1
+          IF (j > lFunc) EXIT
+          c = Func(j:j)
+       ELSE                                                  ! Check for variable
+          isUndef = checkUndef(Func(j:),ib,in)
+          if (isUndef) then
+             j = j+in-1
+             IF (j> lFunc) EXIT
+             c = Func(j:j)
+          else
+             call GetVariables (Func(j:),ib,in)
+             call variables_in_expression%push_back(Func(j+ib-1:j+in-2))
+             j = j+in-1
+             IF (j > lFunc) EXIT
+             c = Func(j:j)
+          end if
+       END IF
+       DO WHILE (c == ')')                                   ! Check for closing parenthesis
+          ParCnt = ParCnt-1
+          IF (ParCnt < 0) THEN
+             CALL ParseErrMsg (j, FuncStr, ipos, 'Mismatched parenthesis')
+             _ASSERT(.FALSE.,'needs informative message')
```

Comment:
IF the "informative message" is generated by the ParseErrMsg() call, shouldn't that go in now?   Otherwise, this seems a bit verbose.

Reply:
Yes, you are right. This was just a copy and paste from CheckSyntax (which I have not looked at in probably 5 years or more). CheckSyntax has the same issue

Reply:
I push an update that eliminated the redundant parseErrMsg calls